pkg/controller/provider/container/ovirt/collector.go

```diff
@@ -143,13 +143,24 @@ func (r *Collector) Version() (major, minor, build, revision string, err error)
 	if err != nil {
 		return
 	}
-	version := strings.Split(system.Product.Version.FullVersion, ".")
+	major, minor, build, revision = r.parseVersion(system.Product.Version.FullVersion)
+	return
+}
+
+func (r *Collector) parseVersion(fullVersion string) (major, minor, build, revision string) {
```

Comment:
it's kind of a utility function and as such we can separate it out from the Collector struct, which will simplify the tests a little bit
```suggestion
func parseVersion(fullVersion string) (major, minor, build, revision string) {
```